src/get-port.ts

```diff
@@ -24,6 +24,11 @@ export async function getPort(
 ): Promise<PortNumber> {
   if (typeof _userOptions === "number" || typeof _userOptions === "string") {
     _userOptions = { port: Number.parseInt(_userOptions + "") || 0 };
+  } else if (
+    typeof _userOptions.port === "number" ||
+    typeof _userOptions.port === "string" // Shouldn't happen, but just in case
+  ) {
+    _userOptions.port = Number.parseInt(_userOptions.port + "") || 0;
```

Comment:
This might be silent ~> random behavior if port is `XYZ` it will become `0` silently